src/functions/memoize.ts

```diff
@@ -0,0 +1,14 @@
+let memoizedValue: Record<string, string> | null = null;
+
+export function memoize(payload: Record<string, string> | null, options = { overwrite: true }) {
+  if (memoizedValue === null || options.overwrite) {
+    memoizedValue = payload;
+  }
+  memoizedValue = Object.assign({}, memoizedValue, payload);
```

Comment:
Object.assign doesn't work (it will not merge properly deep properties). You need to use something like lodash.merge.